main.tf

```
@@ -67,9 +67,12 @@ resource "aws_lb_target_group" "this" {
     unhealthy_threshold = var.health_check.unhealthy_threshold
   }
 
-  stickiness {
-    enabled = false
-    type    = "lb_cookie"
+  dynamic "stickiness" {
+    for_each = var.stickiness != null ? [var.stickiness] : []
```

```suggestion
    for_each = var.stickiness != null ? toset([var.stickiness]) : []
```
I'm pretty sure terraform requires for_each arguments to be maps or sets but this may have changes in more recent versions of the tool

---

@sblack4 this is working fine in versión 1.0.0, but as you said, in the official documentation it says that _The for_each meta-argument accepts a map or a set of strings_ (https://www.terraform.io/language/meta-arguments/for_each#basic-syntax). 

I have changed it with you suggestion.